src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2605,7 +2605,7 @@ private bool IsOperandErrors(CSharpSyntaxNode node, ref BoundExpression operand,
                     if (!operand.HasAnyErrors)
                     {
                         Error(diagnostics, ErrorCode.ERR_LambdaInIsAs, node);
-                        operand = BadExpression(node, operand);
+                        operand = BadExpression(node, operand).MakeCompilerGenerated();
```

Comment:
I wonder if BadExpression should always be marked as compiler-generated. This could wait til post-15.5.

Reply:
> I wonder if BadExpression should always be marked as compiler-generated.

Not really.